File: Dataset/cpp/whilesum.cpp

//  whilesum.cpp  (c) 1997-2002 Kari Laitinen

#include  <iostream.h>
#include  <iomanip.h>

int main()
{
   int  integer_from_keyboard    =  -1 ;
   int  sum_of_integers          =   0 ;

   cout << "\n This program calculates the sum of integers you"
           "\n type in from the keyboard. By entering a zero you"
           "\n can terminate the program. \n\n" ;

   while ( integer_from_keyboard  !=  0 )
   {
      cout  <<  "  Current sum: "  <<  setw( 8 )  
            <<  sum_of_integers
            <<  "    Enter an integer: "  ;
      cin   >>  integer_from_keyboard ;
      sum_of_integers  =  sum_of_integers + integer_from_keyboard ;
   }
}

File: Dataset/cs/Whilesum.cs

//  Whilesum.cs  (c) 2003 Kari Laitinen

using System ;

class Whilesum
{
   static void Main()
   {
      int  integer_from_keyboard   =  -1 ;
      int  sum_of_integers         =   0 ;

      Console.Write( "\n This program calculates the sum of the integers"
                   + "\n you type in from the keyboard. By entering a"
                   + "\n zero you can terminate the program. \n\n" ) ;

      while ( integer_from_keyboard  !=  0 )
      {
         Console.Write( "  Current sum: "
                     +  sum_of_integers.ToString().PadLeft( 8 )
                     +  "    Enter an integer: " ) ;

         integer_from_keyboard  = Convert.ToInt32( Console.ReadLine() );

         sum_of_integers  =  sum_of_integers + integer_from_keyboard ;
      }
   }
}


